File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const map = new Map();
    
    for (let s of strs){
         let count = new Array(26).fill(0);
        for (let c of s){
            count[c.charCodeAt(0)-'a'.charCodeAt(0)] +=1;
        }
    
    const key = count.join(",");

    if (!map.has(key)){
        map.set(key,[]);
        }

    map.get(key).push(s);
    
    }

    return Array.from(map.values());
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ana_map = defaultdict(list)

        for s in strs:
            count = [0] * 26
            for c in s:
                count[ord(c) - ord("a")] += 1
            ana_map[tuple(count)].append(s)

        return list(ana_map.values())